packages/docusaurus/docs/working-docs/50-functional scenarios/20-inference queries/16-training data pollution.md

```diff
@@ -6,23 +6,76 @@ Is there evidence that training data on which this system has been trained has b
 
 ## Schemas used
 
-Links 
+* [data](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/10-data.v1.0.0.schema.yaml)
+* [data pack](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/20-data-pack.v1.0.0.schema.yaml)
+* [config](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/25-config.v1.0.0.schema.yaml) 
+* [identity](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/5-identity.v1.0.0.schema.yaml)
+* [ai system](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/50-ai-system.v1.0.0.schema.yaml)
+* [poisoned attestation](https://github.com/nqminds/Trusted-AI-BOM/blob/training-data-provenance/packages/schemas/src/taibom-schemas/66-polluted_attestation.v1.0.0.schema.yaml)
 
-* x
-* y
-* z
 
 
 
 ## Pseudo code 
 
-description of the algorithm used 
+```python
+def ai_system_has_polluted_data(AISystemId):
+    Attestations = []
+
+    # Step 1: Find all Data Verification Credentials (DataVcIds) linked to the AI System
+    ConfigVcId = db_get_ai_system(AISystemId).config_vc_id
+    WeightsVcId = get_trained_weights(ConfigVcId)
+
+    if not is_weights_data(WeightsVcId):
+        return []  # No polluted attestations if weights data is invalid
+
+    TrainingSystemVcId = get_training_system_for_weights(WeightsVcId)
+    DatapackVcId = get_datapack_for_training_system(TrainingSystemVcId)
+
+    # Step 2: Extract all dataset verification credentials (DataVcIds) from the datapack
+    DataVcIds = extract_data_vcs_from_datapack(DatapackVcId)
+
+    # Step 3: Identify polluted attestations for each DataVcId
+    for DataVcId in DataVcIds:
+        attestations = get_polluted_attestations_for_data(DataVcId)
+        for attestation in attestations:
+            if attestation.type == "polluted":
+                ComponentHash = attestation.component_hash
+                PollutionDetails = attestation.details
+                Attestations.append(({"component": ComponentHash, "data_vc_id": DataVcId}, PollutionDetails))
+
+    # Step 4: Return all polluted attestations found
+    return Attestations
+```

```

Comment:
Similar to previous, is this implemented ? or should we remove the python syntax so it is purely pseudocode ?